classes/range.js

```diff
@@ -66,14 +67,19 @@ class Range {
       }
     }
 
-    this.format()
+    this.formatted = undefined
+  }
+
+  get range () {
+    if (this.formatted === undefined) {
+      this.formatted = this.set
+        .map((comps) => comps.join(' ').trim())
+        .join('||')
```

Comment:
How often is `range` usually called, would it make sense to replace this with a for loop too?

Reply:
That's a good question! The benchmarks don't seem to contain anything that get the `.range` property. Transforming this into loops like this seems to speed up the calculation considerably though:

```js
  get range () {
    if (this.formatted === undefined) {
      let formatted = ''
      for (let i = 0; i < this.set.length; i++) {
        if (i > 0) {
          formatted += '||'
        }
        const comps = this.set[i]
        for (let k = 0; k < comps.length; k++) {
          if (k > 0) {
            formatted += ' '
          }
          formatted += comps[k].toString().trim()
        }
      }
      this.formatted = formatted
    }
    return this.formatted
  }
```

Reply:
I added this in a new commit that's applied before the lazy calculation commit. By itself this change seemed to boost bench-satisfies and bench-subset scores by up to 20%. Thanks for the suggestion! 👍

Reply:
Nice win!